addons/etcd/scripts/start.sh

```diff
@@ -0,0 +1,47 @@
+echo "start etcd..."
+
+
+export conf=/etc/etcd/etcd.conf
+export tmpconf=/tmp/etcd/etcd.conf
+
+cp $conf $tmpconf
+
+# peer tls check
+initial_advertise_peer_urls=$(sed -n 's/^initial-advertise-peer-urls: //p' $tmpconf)
+peer_protocol=""
+if echo "$initial_advertise_peer_urls" | grep -q "https" ; then
+    peer_protocol="https"
+else
+    peer_protocol="http"
+fi
+
+# when a member joins, initial-cluster needs to be reset because configmap will not update automatically
+MY_PEER=${KB_POD_FQDN}${CLUSTER_DOMAIN}
+PEERS=""
+DOMAIN=$KB_NAMESPACE".svc"$CLUSTER_DOMAIN
+i=0 
+while [ $i -lt $KB_REPLICA_COUNT ]; do
+    if [ $i -ne 0 ]; then
+        PEERS="$PEERS,";
+    fi; 
+    host=$(eval echo \$KB_"$i"_HOSTNAME)
+    host=$host"."$DOMAIN
```

Comment:
$KB_"$i"_HOSTNAME is deprecated, could use vars.componentVarRef.podFQDNs

Reply:
after configmap dynamic rerendering hscale, this part can be deleted